src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java

```diff
@@ -302,6 +310,36 @@ public void validateOptions() throws ConfigurationException
         }
     }
 
+    @Override
+    public void maybeWarnOnOptions()
+    {
+        if (!SchemaConstants.isSystemKeyspace(keyspaceName))
+        {
+            ImmutableMultimap<String, InetAddressAndPort> dcsNodes = StorageService.instance.getTokenMetadata()
+                                                                                            .getDC2AllEndpoints(snitch);
+            for (Entry<String, String> e : this.configOptions.entrySet())
+            {
+
+                String dc = e.getKey();
+                ReplicationFactor rf = getReplicationFactor(dc);
+                int nodeCount = dcsNodes.get(dc).size();
+                // nodeCount==0 on many tests
```

Comment:
Can you explain, please?

Reply:
Tests that don't set up a complete cluster with network, gossip, etc will return the number of available nodes as '0', which makes sense but only for a testing env where things are half-baked.

Reply:
Btw that is why we require `requireNetwork()` on the newly added test in `AlterTest`. Otherwise the test sees 0 nodes and no warnings are generated.